openedx/core/djangoapps/content_libraries/management/commands/migrate_lib_to_learning_core.py

```diff
@@ -0,0 +1,234 @@
+                )
+            # Publish all the Draft versions we created.
+            publishing_api.publish_all_drafts(
+                learning_package_id=learning_package.id,
+                message="Initial import from Blockstore",
+                published_at=now,
+            )
+
+            # Now grab the draft version from blockstore, and copy those...
+            draft_files = get_bundle_files_dict(lib.bundle_uuid, use_draft=lib_constants.DRAFT_NAME)
+            draft_definition_files = {
+                file_path: metadata
+                for file_path, metadata in draft_files.items()
+                if file_path.endswith("definition.xml")
+            }
+            for file_path, draft_metadata in draft_definition_files.items():
+                published_metadata = published_metadata_dict.get(file_path)
+                if draft_metadata.modified:
+                    block_type, block_id, _def_xml = file_path.split('/')
+                    xml_bytes = get_bundle_file_data(bundle.uuid, file_path, use_draft=lib_constants.DRAFT_NAME)
+                    display_name = extract_display_name(xml_bytes, file_path)
+
+                    # If this is newly created in the draft, we have to create a
+                    # whole new Component...
+                    if published_metadata is None:
+                        component = components_api.create_component(
+                            learning_package_id=learning_package.id,
+                            namespace=COMPONENT_NAMESPACE,
+                            type=block_type,
+                            local_key=block_id,
+                            created=now,
+                            created_by=None,
+                        )
+                        component_pk = component.pk
+                        version_num = 1
+                    # Otherwise, it's just been modified...
+                    else:
+                        component_pk = published_component_pks[file_path]
+                        version_num = 2
+
+                    component_version = components_api.create_component_version(
+                        component_pk=component_pk,
+                        version_num=version_num,
+                        title=display_name,
+                        created=now,
+                        created_by=None,
+                    )
+                    text_content, _created = contents_api.get_or_create_text_content_from_bytes(
+                        learning_package_id=learning_package.id,
+                        data_bytes=xml_bytes,
+                        mime_type=f"application/vnd.openedx.xblock.v1.{block_type}+xml",
+                        created=now,
+                    )
+                    components_api.add_content_to_component_version(
+                        component_version,
+                        raw_content_id=text_content.pk,
+                        key="definition.xml",
```

Comment:
@bradenmacdonald: I just copied the blockstore convention of `definition.xml` here, but I was wondering if I should change it to `source.xml` because it doesn't really represent the definition scope, and I was wondering if it might cause more confusion than it's worth.

Or some other name? @kdmccormick: any thoughts?

Reply:
@ormsbee Good point. I like `source.xml` or `block.xml`

Reply:
Thinking on this a little more, `source.xml` might get confused with Python source code (e.g. if we later allow a `src` dir in there for more convenient editing, as a sibling directory for `static`). So with that in mind, I'll go with `block.xml`.

Thank you!

Reply:
`block.xml` sounds great.